### PR TITLE
fix: circular object RD-1572

### PR DESCRIPTION
**Pull request by doriaviram-lumigo**

### CODE REVIEW

File: src/tools/noCirculars.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const noCirculars = v => {
```
Comment (doriaviram-lumigo) [inline src/tools/noCirculars.js]: Copied (And add some nice tests)

File: src/tools/noCirculars.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      a.b = b;
+      let c = noCirculars(a);
+      expect(c).toEqual({
+        b: {
+          a: '[Circular]',
+        },
+      });
+    });
+
+    it('an inherited property', function() {
```
Comment (saartochner) [inline src/tools/noCirculars.test.js]: delete

File: src/tools/noCirculars.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+import { noCirculars } from './noCirculars';
+
+describe('noCirculars', function() {
+  beforeEach(function() {
```
Comment (saartochner) [inline src/tools/noCirculars.test.js]: delete

File: src/tools/noCirculars.test.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      }
+      Child.prototype = new Base();
+
+      let z = noCirculars(new Child());
+      expect(z).toHaveProperty('child', true);
+      expect(z).not.toHaveProperty('base');
+    });
+
+    it('an array-like object', function() {
+      let t = { length: 3, 0: 'test', 1: 'test', 2: 'test' };
```
Comment (saartochner) [inline src/tools/noCirculars.test.js]:
```suggestion
      let t = ['test', 'test2', 'test3'];
```
and check a circular example

**Pull request merged**